### DISCUSSION TITLE
Alternatives to uglify-js for rehype-minify

### DISCUSSION THREAD

**Discussion by shellscape** [rehypejs/rehype#164] [Ideas]
Since uglify has significant issues with a number of bundlers including ESBuild https://github.com/mishoo/UglifyJS/issues/5764, which means issues running in environments like Lambda, I wanted to pop in to see if there's any openness to swapping out uglify for something a bit more modern - perhaps terser. This is something that I would be willing to contribute, but wanted to gauge how open the project is to that.

**Comment by ChristianMurphy** [comment]
Hey @shellscape 👋 

Speaking for myself, I'm open to it.
A few options come to mind.
* https://terser.org/
* https://babeljs.io/docs/babel-preset-minify
* https://swc.rs/docs/configuration/minification
* https://esbuild.github.io/api/#minify

unified projects tend to be node first, so an option like swc or esbuild could be on the table.
Though a JavaScript first implementation, while slower, could make browser support a bit easier, without needing wasm.

**Reply by remcohaszing** [reply]
[Babel minify](https://github.com/babel/minify) is still in beta and hasn’t been updated in two years. I wouldn’t bet on that. UglifyJS [doesn’t have a lot activity](https://github.com/mishoo/UglifyJS/graphs/contributors) either. IMO terser is the safest bet.

**Reply by wooorm** [reply]
Yes, I also like terser. As far as I know it's well maintained. And JS only. And I believe it works in browsers!

**Reply by wooorm** [reply]
Anyway, yeah: definitely open! I think terser is a good tool to use. If someone wants to work on it, :+1:
